src/main/java/com/leverx/learningmanagementsystem/btp/registry/service/impl/CloudSubscriptionService.java

```diff
@@ -0,0 +1,118 @@
+package com.leverx.learningmanagementsystem.btp.registry.service.impl;
+
+import com.leverx.learningmanagementsystem.btp.destinationservice.config.DestinationServiceConfiguration;
+import com.leverx.learningmanagementsystem.btp.registry.model.DependenciesResponseDto;
+import com.leverx.learningmanagementsystem.btp.registry.service.SubscriptionService;
+import com.leverx.learningmanagementsystem.btp.servicemanager.dto.BindResource;
+import com.leverx.learningmanagementsystem.btp.servicemanager.dto.CreateSchemaDto;
+import com.leverx.learningmanagementsystem.btp.servicemanager.dto.Parameters;
+import com.leverx.learningmanagementsystem.btp.servicemanager.dto.SchemaBindingRequest;
+import com.leverx.learningmanagementsystem.btp.servicemanager.dto.SchemaBindingResponse;
+import com.leverx.learningmanagementsystem.btp.servicemanager.dto.SchemaInstanceResponse;
+import com.leverx.learningmanagementsystem.btp.servicemanager.service.ServiceManager;
+import com.leverx.learningmanagementsystem.core.app.config.AppConfiguration;
+import com.leverx.learningmanagementsystem.db.service.dbmigrator.DataBaseMigrator;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.leverx.learningmanagementsystem.btp.registry.constants.RegistryConstants.BINDING;
+import static com.leverx.learningmanagementsystem.btp.registry.constants.RegistryConstants.DB_NAME;
+import static com.leverx.learningmanagementsystem.btp.registry.constants.RegistryConstants.ROUTER_URL;
+import static com.leverx.learningmanagementsystem.btp.registry.constants.RegistryConstants.SCHEMA;
+import static com.leverx.learningmanagementsystem.btp.registry.constants.RegistryConstants.SERVICE_PLAN_ID;
+import static com.leverx.learningmanagementsystem.btp.registry.constants.RegistryConstants.TENANT_ID;
+
+@Service
+@Profile("cloud")
+@Slf4j
+@AllArgsConstructor
+public class CloudSubscriptionService implements SubscriptionService {
+
+    private final ServiceManager serviceManager;
+    private final AppConfiguration appConfiguration;
+    private final DataBaseMigrator dataBaseMigrator;
+    private final DestinationServiceConfiguration destinationConfiguration;
+
+    @Override
+    public String subscribe(String tenantId, String tenantSubDomain) {
+        CreateSchemaDto createSchemaDto = configureCreateSchemaDto(tenantId);
+
+        log.info("Assigning schema {} to tenant {}", createSchemaDto.name(), tenantId);
+        SchemaInstanceResponse schemaInstance = serviceManager.createServiceInstance(createSchemaDto);
```

Comment:
logic with schema creation is out of scope of this stage and must be removed and added only in related stage